framework/src/mesh/MooseMesh.C

```diff
@@ -215,6 +215,10 @@ MooseMesh::MooseMesh(const InputParameters & parameters)
   else if (isParamValid("block"))
     _provided_coord_blocks = getParam<std::vector<SubdomainName>>("block");
 
+  if (getParam<bool>("build_all_side_lowerd_mesh"))
+    // Do not initially allow removal of remote elements
+    allowRemoteElementRemoval(false);
```

Comment:
Where are we then restoring `allowRemoteElementRemoval(true)`?

Reply:
Currently nowhere. But this method marks `_need_delete = true` if someone calls with `allowRemoteElementRemoval(false)`, and that flag is checked during the `delete_remote_elements_after_late_geometric_ghosting` MOOSE action task.